namesilo.php

```diff
@@ -2266,91 +2580,75 @@ private function manageSettings(
+        if (isset($checkDomainStatus)) {
+            return $checkDomainStatus;
+        }
 
-            $this->view = new View($view, 'default');
-            // Load the helpers required for this view
-            Loader::loadHelpers($this, ['Form', 'Html']);
+        $this->view = new View($view, 'default');
+        // Load the helpers required for this view
+        Loader::loadHelpers($this, ['Form', 'Html']);
 
-            $row = $this->getModuleRow($package->module_row);
-            $api = $this->getApi($row->meta->user, $row->meta->key, $row->meta->sandbox == 'true');
-            $domains = new NamesiloDomains($api);
-            $transfer = new NamesiloDomainsTransfer($api);
+        $row = $this->getModuleRow($package->module_row);
+        $api = $this->getApi($row->meta->user, $row->meta->key, $row->meta->sandbox == 'true');
+        $domains = new NamesiloDomains($api);
+        $transfer = new NamesiloDomainsTransfer($api);
 
-            $fields = $this->serviceFieldsToObject($service->fields);
+        $fields = $this->serviceFieldsToObject($service->fields);
 
-            if (!empty($post)) {
-                if (isset($post['resend_verification_email'])) {
-                    $response = $domains->emailVerification(['email' => $post['resend_verification_email']]);
+        if (!empty($post)) {
+            if (isset($post['resend_verification_email'])) {
+                $response = $domains->emailVerification(['email' => $post['resend_verification_email']]);
+                $this->processResponse($api, $response);
+            } else {
+                if (isset($post['registrar_lock'])) {
+                    $LockAction = $post['registrar_lock'] == 'Yes' ? 'Lock' : 'Unlock';
+                    $response = $domains->setRegistrarLock($LockAction, ['domain' => $fields->domain]);
                     $this->processResponse($api, $response);
-                } else {
-                    if (isset($post['registrar_lock'])) {
-                        $LockAction = $post['registrar_lock'] == 'Yes' ? 'Lock' : 'Unlock';
-                        $response = $domains->setRegistrarLock($LockAction, ['domain' => $fields->domain]);
-                        $this->processResponse($api, $response);
-                    }
-
-                    if (isset($post['request_epp'])) {
-                        $response = $transfer->getEpp(['domain' => $fields->domain]);
-                        $this->processResponse($api, $response);
-                        unset($post['request_epp']);
-                        $this->setMessage(
-                            'success',
-                            Language::_(
-                                'Namesilo.!success.epp_code_sent',
-                                true
-                            )
-                        );
-                    }
-
-                    if (isset($post['whois_privacy_before']) && isset($post['whois_privacy'])) {
```

Comment:
I do not completely understand this, Do you want me to put the Whois Privacy checkbox back on the Settings tab?

Reply:
Yes.  As per https://github.com/blesta/module-namesilo/pull/54#discussion_r712573026 I think the new tab is not necessary.

Reply:
But in this case, how do I limit access? If I move the checkbox back to the Settings tab, it would have the EPP code and the Whois Privacy on the same Tab. For example, what would happen if the service has the EPP Code Option but no ID Protection?

Reply:
Determine whether it can be accessed, store it in a variable, and base display and submission on that variable.